src/nnet3/nnet-simple-component.cc

```diff
@@ -588,34 +608,64 @@ std::string ClipGradientComponent::Info() const {
          << ", clipping-threshold=" << clipping_threshold_
          << ", clipped-proportion="
          << (count_ > 0 ? static_cast<BaseFloat>(num_clipped_)/count_ : 0);
+  if (self_repair_clipped_proportion_threshold_ != BaseFloat(kUnsetValue))
+    stream << ", self-repair-clipped-proportion-threshold="
+           << self_repair_clipped_proportion_threshold_;
+  if (self_repair_clipping_to_value_ != BaseFloat(kUnsetValue))
+    stream << ", self-repair-clipping-to-value="
+           << self_repair_clipping_to_value_;
+  if (self_repair_scale_ != 0.0)
+    stream << ", self-repair-scale=" << self_repair_scale_;
   return stream.str();
 }
 
 void ClipGradientComponent::Init(int32 dim,
                                  BaseFloat clipping_threshold,
                                  bool norm_based_clipping,
+                                 BaseFloat self_repair_clipped_proportion_threshold,
+                                 BaseFloat self_repair_clipping_to_value,
+                                 BaseFloat self_repair_scale,
                                  int32 num_clipped,
-                                 int32 count)  {
+                                 int32 count,
+                                 int32 num_self_repaired,
+                                 int32 num_backproped)  {
   KALDI_ASSERT(clipping_threshold >= 0 && dim > 0);
   dim_ = dim;
   norm_based_clipping_ = norm_based_clipping;
   clipping_threshold_ = clipping_threshold;
+  self_repair_clipped_proportion_threshold_ =
+      self_repair_clipped_proportion_threshold;
+  self_repair_clipping_to_value_ = self_repair_clipping_to_value;
+  self_repair_scale_ = self_repair_scale;
   num_clipped_ = num_clipped;
   count_ = count;
+  num_self_repaired_ = num_self_repaired;
+  num_backproped_ = num_backproped;
 }
 
 void ClipGradientComponent::InitFromConfig(ConfigLine *cfl) {
   int32 dim = 0;
   bool ok = cfl->GetValue("dim", &dim);
   bool norm_based_clipping = false;
   BaseFloat clipping_threshold = 15.0;
+  BaseFloat self_repair_clipped_proportion_threshold = kUnsetValue;
+  BaseFloat self_repair_clipping_to_value = kUnsetValue;
```

Comment:
I think self_repair_clipping_to_value is badly named.  How about self_repair_target?